### PR TITLE
Prompt for RelativeFilePaths in interactive update

### LINKED ISSUE #414: wingetcreate update fails while there is a version number in the zip archive
### Brief description of your issue

The zip archive file `tssh_0.1.8_windows_x86_64.zip` having `tssh_0.1.8_windows_x86_64\tssh.exe` in it. 

The version number `0.1.8` causes `wingetcreate update` to fail.

### Steps to reproduce

```
wingetcreate update --urls `
https://github.com/trzsz/trzsz-ssh/releases/download/v0.1.8/tssh_0.1.8_windows_x86_64.zip `
https://github.com/trzsz/trzsz-ssh/releases/download/v0.1.8/tssh_0.1.8_windows_aarch64.zip `
https://github.com/trzsz/trzsz-ssh/releases/download/v0.1.8/tssh_0.1.8_windows_i386.zip `
--version 0.1.8 trzsz.tssh
```

### Expected behavior

wingetcreate update success.

### Actual behavior

```
Retrieving latest manifest for trzsz.tssh
Downloading and parsing: https://github.com/trzsz/trzsz-ssh/releases/download/v0.1.8/tssh_0.1.8_windows_x86_64.zip...
Nested installer not found in zip archive: tssh_0.1.7_windows_x86_64\tssh.exe
````

### Environment

```shell
Windows Package Manager Manifest Creator v1.2.8.0
```

```
Windows Package Manager v1.5.1881
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.1992
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.20.1881.0

Winget Directories
-------------------------------------------------------------------------------------------------------------------------------
Logs                               %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
User Settings                      %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json
Portable Links Directory (User)    %LOCALAPPDATA%\Microsoft\WinGet\Links
Portable Links Directory (Machine) C:\Program Files\WinGet\Links
Portable Package Root (User)       %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portable Package Root              C:\Program Files\WinGet\Packages
Portable Package Root (x86)        C:\Program Files (x86)\WinGet\Packages

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale

Admin Setting                             State
--------------------------------------------------
LocalManifestFiles                        Enabled
BypassCertificatePinningForMicrosoftStore Disabled
InstallerHashOverride                     Disabled
LocalArchiveMalwareScanOverride           Disabled
```
```

### LINKED ISSUE #430: Prompt for RelativeFilePaths if client cannot auto-detect
### Description of the new feature / enhancement

If we cannot find a match for RelativeFilePath in the new archive in interactive update, the client should ask the user to disambiguate and provide the new RelativeFilePaths.

Example manifests that are applicable:

1.
```powershell
wingetcreate update PyPy.PyPy.3.9 -v 18.0 --interactive
```

Use the URL `https://downloads.python.org/pypy/pypy3.10-v7.3.12-win64.zip` instead of the existing one.

2.
```powershell
wingetcreate update DynamoRIO.DynamoRIO  -v 18.0 --interactive
```
Use the URL `https://github.com/DynamoRIO/dynamorio/releases/download/cronbuild-9.93.19557/DynamoRIO-Windows-9.93.19557.zip` instead of the existing one.

3.
```powershell
wingetcreate update ZakKemble.avr-gcc -v 18.0 --interactive
```

Use the URL `https://github.com/ZakKemble/avr-gcc-build/releases/download/v11.1.0-1/avr-gcc-11.1.0-x64-windows.zip` instead of old one

The point of using different URLs is so that we can get different RelativeFilePaths than the existing ones to test the behavior.

### Proposed technical implementation details

_No response_

### PR DESCRIPTION
**Pull request by mdanish-kh**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Are you working against an Issue?
	- Resolves #430

	
Examples manifests to test with are included in the linked issue

Unmatched RelativeFilePaths can be categorized into 2 types with this PR

#### a. Single match for `example.exe` in the new archive
e.g. `Example-1.2\example.exe` -> `Example-1.3\example.exe`

In this case the autonomous and interactive update logic will auto-detect the new binary from the archive and user does not have to do anything.

#### b. Multiple matches for `example.exe` in the new archive
e.g. `Example-1.2\win64\example.exe` matches with `Example-1.3\win64\example.exe` and `Example-1.3\win32\example.exe` in the new archive.

In this case the autonomous update will fail as it cannot disambiguate which `example.exe` to use as the new RelativeFilePath (although the above example makes it seem obvious which to choose but we need a way to generalize for any case). Interactive update will match for all `example.exe` in the new archive and prompt the user to select one

### CASE-II Different binary name under different directories
 
e.g. `Example-1.2\example-1.2.exe` -> `Example-1.3\example-1.3.exe`

Autonomous update will fail again for same reasons. Interactive update will now scan for all `.exe` (old relativefilepath extension) in the new archive and prompt the user to select one of them as new RelativeFilePath. If there are no `.exe` files in the new archive, the interactive update can't move any forward and an appropriate message is printed.

-----

 ###### Microsoft Reviewers: codeflow:open?pullrequest=https://github.com/microsoft/winget-create/pull/431&drop=dogfoodAlpha

### CODE REVIEW

File: src/WingetCreateCLI/Commands/UpdateCommand.cs [hunk 1]
```diff
@@ -781,33 +806,26 @@ private async Task UpdateSingleInstallerInteractively(Installer installer)
...
-                            }
-                        }
+                    if (isRelativePathNull)
+                    {
+                        Logger.ErrorLocalized(nameof(Resources.NestedInstallerTypeNotFound_Error));
+                        continue;
                     }
+
+                    string pathToNestedInstaller = Path.Combine(extractDirectory, installer.NestedInstallerFiles.First().RelativeFilePath);
+                    packageFile = pathToNestedInstaller;
```
Comment (ryfu-msft) [inline src/WingetCreateCLI/Commands/UpdateCommand.cs:828]: Can assign directly to packageFile

File: src/WingetCreateCLI/Properties/Resources.resx [hunk 2]
```diff
@@ -1031,6 +1031,18 @@
     <value>The url of the hosted icon file</value>
   </data>
   <data name="MultipleMatchingNestedInstallersFound_Error" xml:space="preserve">
-    <value>Multiple matches found for "{0}" from {1}</value>
+    <value>Multiple matches found for "{0}" from {1}. Please use the interactive mode to update this package.</value>
```
Comment (ryfu-msft) [inline src/WingetCreateCLI/Properties/Resources.resx:1034]: Need an explanation for `{0}` and `{1}` similar to what is done on line 968. Same for the ones below.

**Review by ryfu-msft — CHANGES_REQUESTED** [review]

File: src/WingetCreateCLI/Commands/UpdateCommand.cs [hunk 3]
```diff
@@ -622,6 +599,54 @@ private static void DisplayManifestsAsMenuSelection(Manifests manifests)
...
+                    if (sameExtensionFiles.Count > 0)
+                    {
+                        Console.WriteLine();
+                        Logger.WarnLocalized(nameof(Resources.NestedInstallerFileNotFound_Warning), oldRelativeFilePath);
+                        return Prompt.Select(Resources.SelectRelativeFilePath_Message, sameExtensionFiles);
+                    }
+                    else
+                    {
+                        return null;
+                    }
```
[the remark this answers is not in the document]
Comment (mdanish-kh) [reply src/WingetCreateCLI/Commands/UpdateCommand.cs:642]: The error below is not a good explanation of what is happening, it says `...Please use the interactive update to update this package` (as it's meant for autonomous update scenario), but the user is already using the interactive update and the actual error is something else. I return null here and then below I check for `isRelativePathNull` and display an appropriate error message there 

```
                    if (isRelativePathNull)
                    {
                        Logger.ErrorLocalized(nameof(Resources.NestedInstallerTypeNotFound_Error));
                        continue;
                    }
```

**Review by ryfu-msft — APPROVED** [review]

**Pull request merged**